CompoundV3-Bots/Governance-Events/src/agent.spec.ts

```diff
@@ -0,0 +1,375 @@
+import {
+  ethers,
+  Finding,
+  FindingSeverity,
+  FindingType,
+  HandleTransaction,
+  Network,
+} from "forta-agent";
+import { MockEthersProvider, TestTransactionEvent } from "forta-agent-tools/lib/test";
+import { createChecksumAddress, NetworkManager } from "forta-agent-tools";
+
+import { provideInitialize, provideHandleTransaction } from "./agent";
+import { AgentConfig, NetworkData } from "./utils";
+import { APPROVAL_ABI, PAUSE_ACTION_ABI, WITHDRAW_RESERVES_ABI } from "./constants";
+
+function createPauseActionFinding(
+  comet: string,
+  supplyPaused: boolean,
+  transferPaused: boolean,
+  withdrawPaused: boolean,
+  absorbPaused: boolean,
+  buyPaused: boolean,
+  chainId: number
+): Finding {
+  return Finding.from({
+    name: "Pause action on Comet contract",
+    description: "A pause action was executed in a Comet contract",
+    alertId: "COMP2-2-1",
+    protocol: "Compound",
+    type: FindingType.Info,
+    severity: FindingSeverity.Medium,
+    metadata: {
+      chain: Network[chainId],
+      comet,
+      supplyPaused: supplyPaused.toString(),
+      transferPaused: transferPaused.toString(),
+      withdrawPaused: withdrawPaused.toString(),
+      absorbPaused: absorbPaused.toString(),
+      buyPaused: buyPaused.toString(),
+    },
+    addresses: [comet],
+  });
+}
+
+function createWithdrawReservesFinding(
+  comet: string,
+  to: string,
+  amount: ethers.BigNumberish,
+  chainId: number
+): Finding {
+  return Finding.from({
+    name: "Withdraw reserves action on Comet contract",
+    description: "A withdraw reserves action was executed in a Comet contract",
+    alertId: "COMP2-2-2",
+    protocol: "Compound",
+    type: FindingType.Info,
+    severity: FindingSeverity.High,
+    metadata: {
+      chain: Network[chainId],
+      comet,
+      to,
+      amount: amount.toString(),
+    },
+    addresses: [comet, to],
+  });
+}
+
+function createApproveFinding(
+  comet: string,
+  token: string,
+  spender: string,
+  amount: ethers.BigNumberish,
+  chainId: number
+): Finding {
+  return Finding.from({
+    name: "Token approval from Comet contract",
+    description: "A token approval was emitted from a Comet contract",
+    alertId: "COMP2-2-3",
+    protocol: "Compound",
+    type: FindingType.Info,
+    severity: FindingSeverity.High,
+    metadata: {
+      chain: Network[chainId],
+      comet,
+      token,
+      spender,
+      amount: amount.toString(),
+    },
+    addresses: [token, comet, spender],
+  });
+}
+
+const addr = createChecksumAddress;
+const COMET_ADDRESSES = [addr("0x1"), addr("0x2")];
+const TOKEN_ADDRESS = addr("0x3");
+const IRRELEVANT_ADDRESS = addr("0x4");
+
+const network = Network.MAINNET;
+
+const DEFAULT_CONFIG: AgentConfig = {
+  [network]: {
+    cometAddresses: COMET_ADDRESSES,
+  },
+};
+
+describe("Bot Test Suite", () => {
```

Comment:
Just a small detail, I think it's better to include the bot name here for a more specific description, wdyt?

Reply:
Sure, agreed, done on [6b2edcb](https://github.com/NethermindEth/Forta-Agents/pull/371/commits/6b2edcb19f2a0dc726cbba59dd9b606c875c8b37).